src/core/thread/mle_router.cpp

```diff
@@ -2724,7 +2724,7 @@ otError MleRouter::SendChildIdResponse(Child *aChild)
     SuccessOrExit(error = AppendActiveTimestamp(*message));
     SuccessOrExit(error = AppendPendingTimestamp(*message));
 
-    if (aChild->GetState() != Neighbor::kStateValid)
+    if (aChild->GetRloc16() == 0)
```

Comment:
A Child should never have an RLOC16 = 0.  Child ID = 0 is reserved for the Router.